File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/DomainOrderRequest.js
goog.provide('API.Client.DomainOrderRequest');

/**
 * Request body for the domain order flow: putDomains (returns required fields), patchDomains (validates fields), addDomain (places the order). The TLD is resolved from hostname. Per-TLD registrant/contact fields returned by putDomains may be supplied as additional properties on patchDomains/addDomain.
 * @record
 */
API.Client.DomainOrderRequest = function() {}

/**
 * Fully-qualified domain to register or transfer (e.g. example.com).
 * @type {!string}
 * @export
 */
API.Client.DomainOrderRequest.prototype.hostname;

/**
 * Order type.
 * @type {!string}
 * @export
 */
API.Client.DomainOrderRequest.prototype.type;

/**
 * Coupon code (addDomain only).
 * @type {!string}
 * @export
 */
API.Client.DomainOrderRequest.prototype.coupon;

/**
 * Set to \"enable\" to add Whois privacy (addDomain only).
 * @type {!string}
 * @export
 */
API.Client.DomainOrderRequest.prototype.whoisPrivacy;

/** @enum {string} */
API.Client.DomainOrderRequest.TypeEnum = { 
  register: 'register',
  transfer: 'transfer',
}
/** @enum {string} */
API.Client.DomainOrderRequest.WhoisPrivacyEnum = { 
  enable: 'enable',
  disable: 'disable',
}

File: OPENAPI-REST-API/openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/DomainOrderRequest.scala

package org.openapitools.client.model

import java.util.HashMap

case class DomainOrderRequest (
    /* Fully-qualified domain to register or transfer (e.g. example.com). */
    _hostname: String,
    /* Order type. */
    _type: Option[String],
    /* Coupon code (addDomain only). */
    _coupon: Option[String],
    /* Set to \"enable\" to add Whois privacy (addDomain only). */
    _whois_privacy: Option[String]
)
object DomainOrderRequest {
    def toStringBody(var_hostname: Object, var_type: Object, var_coupon: Object, var_whois_privacy: Object) =
        s"""
        | {
        | "hostname":$var_hostname,"type":$var_type,"coupon":$var_coupon,"whois_privacy":$var_whois_privacy
        | }
        """.stripMargin
}
